### PR TITLE
Onboardjs Integration

### PR DESCRIPTION
**Pull request by lovrobiljeskovic**
- Added `onboardjs` lib
- Removed `use-wallet` lib
- Created new `OnboardProvider`
- Refactoring all the components which used `use-wallet` lib
- Removed unnecessary complexity from Providers and UI
- updated `localStorage`'s migration function and added a new version
- Ensuring all wallet login/signup flows are in order

### CODE REVIEW

**Review by JamesLefrere — CHANGES_REQUESTED** [review]
This is a task that touches lots of parts of the app so there are lots of details to attend to, and the quality level _must_ be very high because the wallet connection is a critical part of the apps. Therefore, I'm giving very detailed feedback to help us reach the quality level required.

There are a number of things that must be taken care of before I can sign off on this, but taking care of them will make sure that it's as robust and simple as possible. It's mainly centred around the onboard provider and ensuring that the UX is the same where possible. 

This issue's 'Definition of Done' is:

- All wallet/tx interaction works (this seems to work with MetaMask :+1: )
- The migration works (if you need to test it, go to devtools -> application -> local storage, and set version to 2)
- Automatic reconnecting works, and users don't see the modal when reconnecting
- Clicking on forms when disconnected opens the modal rather than an empty account screen
- Ideally, the modal slightly styled (I need to provide a visual hint for that), or at least the blocknative text is hidden
- The wallets with extra login/signup processes don't cause unexpected behaviour, or the state is at least recoverable if the user cancels signup

File: CHANGELOG.md
Comment (lovrobiljeskovic) [reply CHANGELOG.md:5]: Don't tell me twice!

File: src/web3/connectors.ts
Comment (lovrobiljeskovic) [reply src/web3/connectors.ts:78]: Actually it's used here:

```
const migrateLocalStorage = (): void => {
  const version = LocalStorage.get('version');

  if (version === 0) {
    const connectorId = LocalStorage.get<'connectorId'>('connectorId');

    if (connectorId) {
      // The injected connector now requires a subType (MetaMask/Brave/etc),
      // so we can't assume the type; these users will have to connect again.
      if (connectorId !== 'injected') {
        const subType = CONNECTORS.find(c => c.id === connectorId)?.subType;
        LocalStorage.set('connector', {
          id: connectorId,
          subType,
        });
      }
      LocalStorage.removeItem<'connectorId'>('connectorId');
    }

    LocalStorage.setVersion(1);
  }
```

Do we have to replace the `CONNECTORS` with `WALLETS` here? I assumed since this is an old version of local storage, we still have to reference the `CONNECTORS`. Curios if this is true

File: src/components/layout/AppBar.tsx
Comment (lovrobiljeskovic) [reply src/components/layout/AppBar.tsx:206]: Ahh so I guess this explains my above comment? :thinking:

File: src/context/OnboardProvider.tsx
Comment (lovrobiljeskovic) [reply src/context/OnboardProvider.tsx:65]: Do you mean this: 
```export const useOnboard = (): State['onboard'] => useContext(context).onboard;``` ?

File: src/context/OnboardProvider.tsx
Comment (lovrobiljeskovic) [reply src/context/OnboardProvider.tsx:42]: Yeah you are right, it was done because of this

File: src/context/OnboardProvider.tsx
Comment (lovrobiljeskovic) [reply src/context/OnboardProvider.tsx:119]: Thanks for the tip here, I was kinda sure that this might not be the right way to do it!

File: src/context/onboardUtils.tsx
Comment (lovrobiljeskovic) [reply src/context/onboardUtils.tsx:13]: `preferred` ones just show up first, then for the rest you gotta click `show more` so yeah I will just add these as `preferred`

File: src/web3/connectors.ts
Comment (JamesLefrere) [reply src/web3/connectors.ts:78]: Oh ok, then we can leave it in :+1:

**Review by JamesLefrere — APPROVED** [review]

**Comment by JamesLefrere** [comment]
This is looking good now, but holding off on merging while we try to break it on staging.

**Pull request merged**
